services/user-svc/grpcserver/dependencies.go

```diff
@@ -22,9 +22,7 @@ func newDependencies(ctx context.Context, conf config.Config) (*dependencies, er
 	var deps dependencies
 	switch t := conf.Datastore.(type) {
 	case config.LocalDatastore:
-		deps = dependencies{
-			userRepo: local.NewUserRepository(),
-		}
```

Comment:
More future-proof in case we add `deps.onClose` outside of the `switch` in the future; replacing `deps` will cause the outer scoped `deps` to be discarded with the on close handlers.